components/ui/scroll-section-group/use-scroll-section-group.tsx

```diff
@@ -0,0 +1,66 @@
+import { useCallback, useRef, useState, createRef } from "react";
+import { ScrollSection, ScrollSectionGroup, ScrollSectionGroupProps } from "./scroll-section-group";
+
+interface MoveScrollSectionGroupToSectionInput<TSectionId extends string> {
+  sectionId: TSectionId;
+}
+
+interface UseScrollSectionGroupInput<TSectionId extends string> {
+  scrollSectionGroupProps: Omit<ScrollSectionGroupProps<TSectionId>, 'currentIndex' | 'setCurrentIndex' | 'sections' | 'hasEnoughTimePassedSinceLastScrollEventActioned' | 'setLastScrollEventActionedUnixEpochTimestampMilliseconds'> & {
+    sections: Omit<ScrollSection<TSectionId>, 'ref'>[];
+  };
+}
+
+interface UseScrollSectionGroupOutput<TSectionId extends string> {
+  renderScrollSectionGroup: () => React.ReactNode;
+  moveScrollSectionGroupToSection: (input: MoveScrollSectionGroupToSectionInput<TSectionId>) => void;
+}
+
+export const useScrollSectionGroup = <TSectionId extends string>(input: UseScrollSectionGroupInput<TSectionId>): UseScrollSectionGroupOutput<TSectionId> => {
+  const [currentIndex, _setCurrentIndex] = useState<number>(0);
+  const sectionRefs = useRef(Array.from({length: input.scrollSectionGroupProps.sections.length}, _ => createRef<HTMLElement>()));
+  const [lastScrollEventActionedUnixEpochTimestampMilliseconds, setLastScrollEventActionedUnixEpochTimestampMilliseconds] = useState<number | undefined>(undefined);
+
+  const hasEnoughTimePassedSinceLastScrollEventActioned = (unixEpochTimestampMilliseconds: number): boolean => {
+    if(!lastScrollEventActionedUnixEpochTimestampMilliseconds) {
+      return true;
+    }
+
+    return unixEpochTimestampMilliseconds - lastScrollEventActionedUnixEpochTimestampMilliseconds > input.scrollSectionGroupProps.scrollThrottleThresholdMilliseconds;
+  }
+
+
+  const setCurrentIndex = (newCurrentIndex: number) => {
+    _setCurrentIndex(newCurrentIndex);
+
+    sectionRefs.current.at(newCurrentIndex)?.current?.scrollIntoView();
+  }
+
+  const renderScrollSectionGroup = useCallback(() => {
+    return <ScrollSectionGroup
+      className={input.scrollSectionGroupProps.className}
+      hasEnoughTimePassedSinceLastScrollEventActioned={hasEnoughTimePassedSinceLastScrollEventActioned}
+      setLastScrollEventActionedUnixEpochTimestampMilliseconds={setLastScrollEventActionedUnixEpochTimestampMilliseconds}
+      scrollThrottleThresholdMilliseconds={input.scrollSectionGroupProps.scrollThrottleThresholdMilliseconds}
+      sections={input.scrollSectionGroupProps.sections.map((section, index) => ({...section, ref:sectionRefs.current[index]}))}
+      currentIndex={currentIndex}
+      setCurrentIndex={setCurrentIndex}
+    />
+  }, [input.scrollSectionGroupProps, currentIndex, setCurrentIndex]);
+
+  const moveScrollSectionGroupToSection = useCallback(({ sectionId }: MoveScrollSectionGroupToSectionInput<TSectionId>) => {
```

Comment:
Basically what I wanted here was the ability to route someone to `http://localhost:3000/pathogen/arbovirus/wip?section=map` or `http://localhost:3000/pathogen/arbovirus/wip?section=table` or `http://localhost:3000/pathogen/arbovirus/wip?section=visualizations` and have those routes immediately scroll you to the right section. If you look at the designs, there are a whole bunch of links to specific sections so I'm not making this up lol 

**I'm not scrolling at all in the gif below, the route parameter is taking me to the right spot on it's own**

![Peek 2024-01-21 16-28](https://github.com/PathoTracker/Pathotracker/assets/86806388/ed261200-a0d7-452b-b6df-324ee859d9d7)